src/main/kotlin/com/bridgecrew/ui/vulnerabilitiesTree/CheckovToolWindowTree.kt

```diff
@@ -82,17 +92,73 @@ class CheckovToolWindowTree(val project: Project, val split: JBSplitter, private
             }
         }
 
+        expandsFromPathState(tree, CheckovGlobalState!!.expandedDescendants)
+
+        tree.addTreeExpansionListener(object : TreeExpansionListener {
+            override fun treeExpanded(event: TreeExpansionEvent?) {
+                CheckovGlobalState!!.expandedDescendants = getExpandedDescendants((event?.source as Tree))
```

Comment:
sure we need the !! when it's an object? just making sure it doesn't work without